python/ray/experimental/client/server/server.py

```diff
@@ -23,19 +26,87 @@ def __init__(self, test_mode=False):
         self.registered_actor_classes = {}
         self._test_mode = test_mode
 
+    def ClusterInfo(self, request,
+                    context=None) -> ray_client_pb2.ClusterInfoResponse:
+        resp = ray_client_pb2.ClusterInfoResponse()
+        resp.type = request.type
+        if request.type == ray_client_pb2.ClusterInfoType.CLUSTER_RESOURCES:
+            resources = ray.cluster_resources()
+            # Normalize resources into floats
+            # (the function may return values that are ints)
+            float_resources = {k: float(v) for k, v in resources.items()}
+            resp.resource_table.CopyFrom(
+                ray_client_pb2.ClusterInfoResponse.ResourceTable(
+                    table=float_resources))
+        elif request.type == ray_client_pb2.ClusterInfoType.AVAILABLE_RESOURCES:
+            resources = ray.available_resources()
+            # Normalize resources into floats
+            # (the function may return values that are ints)
+            float_resources = {k: float(v) for k, v in resources.items()}
+            resp.resource_table.CopyFrom(
+                ray_client_pb2.ClusterInfoResponse.ResourceTable(
+                    table=float_resources))
+        else:
+            resp.json = self._return_debug_cluster_info(request, context)
+        return resp
+
+    def _return_debug_cluster_info(self, request, context=None) -> str:
+        data = None
+        if request.type == ray_client_pb2.ClusterInfoType.NODES:
+            data = ray.nodes()
+        elif request.type == ray_client_pb2.ClusterInfoType.IS_INITIALIZED:
+            data = ray.is_initialized()
+        else:
+            raise TypeError("Unsupported cluster info type")
+        return json.dumps(data)
+
+    def Terminate(self, request, context=None):
+        if request.WhichOneof("terminate_type") == "task_object":
+            object_ref = cloudpickle.loads(request.task_object.id)
+            try:
+                obj = self.object_refs.get(object_ref.binary(), None)
+                if obj is None:
+                    return ray_client_pb2.TerminateResponse(ok=False)
+                ray.cancel(
```

Comment:
I'm not sure about this assumption that the object ref to cancel must be in self.object_refs. What if the object is created in a remote actor, and the ref is returned to the client somehow?

Could we remove the check here?